napalm_eos/eos.py

```diff
@@ -1201,3 +1201,162 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
                 previous_probe_ip_address = ip_address
 
         return traceroute_result
+
+    def get_bgp_neighbors_detail(self, neighbor_address=''):
+        """Function to return detailed BGP information.
+
+        Information returned by this function can be about a single peer or
+        about all peers. This can be controlled using the following:
+
+        :params neighbor_address: Retuns the statistics for a specific
+                                  BGP neighbor or for all BGP neighbors.
+        """
+
+        peer_ver = None
+        is_all = True if not neighbor_address else False
+
+        if is_all:
+            command = ['show ip bgp neighbors vrf all']
+        else:
+            peer_ver = IPAddress(neighbor_address).version
```

Comment:
Mmm, your particular example does something undocumented. I missed that one. I think we should just propagate the exception for this particular case as that's something that is consistent across drivers. I agree that for vendor-specific exception we should catch them and generate our own.

What I really don't like is having to find out if there is an error by checking a key on a dictionary. Exceptions are precisely designed for that, for reporting errors:

    result = my_method()
    if error in result:
        #now try to find out the error and handle error
    else:
        #normal code

That code above feels wrong to me:

    try:
       result = my_method()
    except AddrFormatError:
       # handle that
    except SomethingElseError:
       # handle this

feels way nicer to me : )